polkadot/node/network/bridge/src/rx/mod.rs

```diff
@@ -1135,13 +1135,28 @@ async fn dispatch_validation_events_to_all<I>(
 	I: IntoIterator<Item = NetworkBridgeEvent<net_protocol::VersionedValidationProtocol>>,
 	I::IntoIter: Send,
 {
+	macro_rules! send_message {
+		($event:expr, $message:ident) => {
+			if let Ok(event) = $event.focus() {
+				let has_high_priority = matches!(
+					event,
+					NetworkBridgeEvent::PeerConnected(..) | NetworkBridgeEvent::PeerViewChange(..)
```

Comment:
PeerDisconnected needs to be on the priority queue as well, otherwise you could have something like:
1. PeerConnected -> Processed Fast
2. PeerDisconnected -> Processed Later
3. PeerConnected -> Processed Fast.

So, you could end up with processing 2 after 3, which is not what we want.

Reply:
On the same topic, what about OurViewChange ?

Reply:
Seems reasonable to me because we need to have an actual state of peers for proper gossiping.

Reply:
I see that OurViewChange is sent via an unbounded channel which we use for prioritization as well. I think it can be resolved later with https://github.com/paritytech/polkadot-sdk/issues/824